### PR TITLE
EASY-1952: cleanup temporary zip file after failure

### PR DESCRIPTION
**Pull request by rvanheest**
fixes EASY-1952

#### When applied it will
* properly use `Try`, thereby fixing a long outstanding `TODO` (_it was easier to cleanup before fixing the real issue_)
* cleanup temporary zip file after failure

@DANS-KNAW/easy for review

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala [hunk 1]
```diff
@@ -47,30 +47,36 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
...
+  def run(implicit ps: Parameters): Try[URI] = {
+    for {
+      maybeVersionOfId <- bagFacade.getIsVersionOf(ps.bagDir.toPath)
+      _ <- maybeVersionOfId.map(createRefBagsTxt).getOrElse(Success(()))
+      zippedBag <- generateUncreatedTempFile()
+      _ <- zipDir(ps.bagDir, zippedBag)
+      response <- putFile(zippedBag)
+      location <- response.map(handleStorageResponse).tried.flatten
+        .doIfSuccess(_ => zippedBag.delete())
+        .doIfFailure { case _ => zippedBag.delete() }
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: Isn't there a kind of `doAlways` ?

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala [hunk 2]
```diff
@@ -47,30 +47,36 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
...
+    } yield location
+  }
 
-    val zippedBag = generateUncreatedTempFile()
-    zipDir(ps.bagDir, zippedBag)
-    val response = putFile(zippedBag)
+  private def handleStorageResponse(response: CloseableHttpResponse)(implicit ps: Parameters) = {
```
Comment (janvanmansum) [inline src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]:
```suggestion
  private def handleBagStoreResponse(response: CloseableHttpResponse)(implicit ps: Parameters) = {
```

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala (hunk 1 above)
Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: nope, we don't have that implemented.
note that we created `doIfSuccess` and `doIfFailure` ourselves. We hadn't taken this possibility into account. I'd say we leave it as is.

Comment (janvanmansum) [reply src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: OK

**Pull request merged**